core/block_dragger.js

```diff
@@ -0,0 +1,239 @@
+
+goog.require('goog.math.Coordinate');
+goog.require('Blockly.DraggedConnectionManager');
+goog.require('goog.asserts');
+
+Blockly.BlockDragger = function(gesture) {
+  /**
+   * The top block in the stack that is being dragged.
+   * @type {!Blockly.BlockSvg}
+   * @private
+   */
+  this.draggingBlock_ = gesture.startBlock_;
+
+  /**
+   * The workspace on which the block is being dragged.
+   * @type {!Blockly.WorkspaceSvg}
+   * @private
+   */
+  this.workspace_ = gesture.startWorkspace_;
+
+  /**
+   * Object that keeps track of connections on dragged blocks.
+   * @type {!Blockly.DraggedConnectionManager}
+   * @private
+   */
+  this.draggedConnectionManager_ = new Blockly.DraggedConnectionManager(
+      this.draggingBlock_);
+
+  /**
+   * Which delete area the mouse pointer is over, if any.
+   * One of {@link Blockly.DELETE_AREA_TRASH},
+   * {@link Blockly.DELETE_AREA_TOOLBOX}, or {@link Blockly.DELETE_AREA_NONE}.
+   * Only relevant if this is a block drag.
+   * @type {?number}
+   * @private
+   */
+  this.deleteArea_ = null;
+
+  /**
+   * Whether the block would be deleted if dropped immediately.
+   * @type {boolean}
+   * Only relevant if this is a block drag.
+   * @private
+   */
+  this.wouldDeleteBlock_ = false;
+
+  /**
+   * True if there is a {Blockly.BlockDragSurfaceSvg} that can be used to drag
+   * the block.
+   * @type {boolean}
+   * @private
+   */
+  this.useDragSurface_ = this.draggingBlock_.useDragSurface_;
+
+  /**
+   * The drag surface that should be used to drag the block, or null if none is
+   * being used.
+   * @type {BlockDragSurfaceSvg}
+   * @private
+   */
+  this.dragSurface_ = this.useDragSurface_ ?
+      gesture.startWorkspace_.blockDragSurface_ : null;
+
+  /**
+   * The location of the top left corner of the dragging block at the beginning
+   * of the drag.
+   * @type {!goog.math.Coordinate}
+   * @private
+   */
+  this.blockRelativeToSurfaceXY_ = this.draggingBlock_.getRelativeToSurfaceXY();
+
+  /**
+   * The coordinate of the block at the beginning of the drag.
+   * @type {!goog.math.Coordinate}
+   * @private
+   */
+  this.startCoordinate_ = gesture.mouseDownXY;
+};
+
+/**
+ * Start dragging a block.  This includes moving it to the drag surface.
+ */
+Blockly.BlockDragger.prototype.startBlockDrag = function() {
+  // Workspace.startDrag just records the start position, which we already know.
+  // TODO: Figure out where to get the list of bubbles to move when dragging.
+  // TODO: Can setResizesEnabled be done at the same time for both types of drags?
+  this.workspace_.setResizesEnabled(false);
+  // TODO: Add getParent() to blockSvg.
+  if (this.draggingBlock_.parentBlock_) {
+    this.draggingBlock_.unplug();
+    this.draggingBlock_.disconnectUiEffect();
+  }
+  // TODO: Make setDragging_ package.
+  this.draggingBlock_.setDragging_(true);
+  // TODO: Consider where moveToDragSurface should live.
+  this.draggingBlock_.moveToDragSurface_();
```

Comment:
The block knows about the block drag surface--that's why you can call moveToDragSurface and moveOffDragSurface directly on the block.  

The TODO is referring to the fact that I might want to move both of those functions to this class in the future.  Right now I'm leaving them alone because they do some work with the block's internals as well.